### DISCUSSION TITLE
Run a line of bash on login / fail ?

### DISCUSSION THREAD

**Discussion by nez-nix** [dani-garcia/vaultwarden#3144] [Ideas]
Would it be possible to add a docker compose environment variable so that if either of these conditions are triggered:

* User logs in successfuly
* Someone tries but fails password/2FA

So I could have a little bash script in my docker compose directory and send a push notifiaction or other alert every time my vault is accessed?

**Comment by BlackDex** [comment]
I would suggest something like fail2ban for this stuff.
Also, there is an option to require new device mail via this https://github.com/dani-garcia/vaultwarden/blob/9b7e86efc23d6bbb5ee350160085ab8d8421d628/.env.template#LL419C8-L419C8

There also is 2FA mail if that fails. https://github.com/dani-garcia/vaultwarden/blob/main/.env.template#L294..L299

But for all other stuff i would suggest to use something like fail2ban or other tools which read log files.

**Reply by tessus** [reply]
The OP wants rather the admin to receive an email, but your links describe how the user will get an email.

**Reply by BlackDex** [reply]
I nearly describe the built-in options and what they do.
The question from the OP can be handled by fail2ban or other log parsing tools as i also mentioned.

**Reply by nez-nix** [reply]
I think I'm gonna create a simple little log parsing thing for it. It will be a good little project for me.
I've started using [Gotify](https://github.com/gotify)  to get push notifications on my Android phone and I really like it.
That was the main reason I came here to suggest the improvement. I can get a nice little push to my phone when someone logs into my Vaultwarden (or fails!) just by running a curl command.

**Reply by tessus** [reply]
@BlackDex This is true, but fail2ban will only work, if one actually logs to a file. I know a few people who either turn off logging or log to `/dev/null`.

It's also true that vw itself does not support triggering external events. Which is a perfectly valid standpoint, btw. I am just stating a fact. This is not a negative critique.

**Comment by nez-nix** [comment]
OK. I will check those out. Actually, I could maybe put something together myself which watches the log. Thanks.
